hail/src/main/scala/is/hail/expr/ir/MatrixIR.scala

```diff
@@ -842,17 +886,177 @@ case class MatrixAggregateColsByKey(child: MatrixIR, aggIR: IR) extends MatrixIR
 
     val newRVType = typ.rvRowType
     val newColType = typ.colType
+    val newColKeyType = typ.colKeyStruct
+    val newColValueType = typ.colValueStruct
 
     val keyIndices = mv.typ.colKey.map(k => mv.typ.colType.field(k).index)
     val keys = oldColValuesBc.value.map { a => Row.fromSeq(keyIndices.map(a.asInstanceOf[Row].get)) }.toSet.toArray
     val nKeys = keys.length
-    val newColValues = oldColValues.copy(value = keys, t = TArray(newColType))
 
     val keysByColumn = oldColValues.value.map { sa => Row.fromSeq(keyIndices.map(sa.asInstanceOf[Row].get)) }
     val keyMap = keys.zipWithIndex.toMap
     val newColumnIndices = keysByColumn.map { k => keyMap(k) }.toArray
     val newColumnIndicesType = TArray(TInt32())
 
+    // Column aggregations
+
+    val (rvAggsColTemplate, makeInitCol, makeSeqCol, aggResultTypeCol, postAggIRCol) = ir.CompileWithAggregators[Long, Long, Long, Long](
```

Comment:
sorry, having trouble getting my head around this whole block. Is it possible to just use `Interpret` here?

Reply:
Not sure. To me this makes sense, but maybe interpret makes more sense to you since you wrote that ;)

Reply:
I mean it might be possible to use Interpret instead of _all_ of this code